### DISCUSSION TITLE
High number of daily commands in a low use enviromment

### DISCUSSION THREAD

**Discussion by chooban** [taskforcesh/bullmq#2976] [Q&A] [answered]
I've used BullMQ in a small personal project and it works great, thank you! The issue I'm currently having is that it's issuing a lot of daily commands which is pushing up costs which, on a personal project, I want to be as low as possible. For background, it's a bluesky feed generator looking for kickstarter projects, and it's probably over-engineered but that's what my personal projects are for! 😄 

The app is hosted on fly.io which provides redis via upstash.io. I have three queues and two scheduled jobs, and the upstash stats show that it's making 44k commands per day. The scheduled jobs only run once per hour, and the remaining queue really only gets a job to execute every 5-10 minutes at most. 

I can see that there are five connections open (though I thought I'd shared the connection) and it's issuing 4 requests per second (two reads, two writes). 

Is this normal behaviour, or have I likely configured something wrong?

**Comment by manast** [comment, marked as answer]
As far as I know, Upstash is now not charging for commands run inside BullMQ's scripts, only charging for 1 call per EVAL call. However, in this case I think the issue is with the "drainDelay" setting.  This setting specifies the duration of the blocking commands in BullMQ, by default 5 seconds, that means that at least one command every 5 seconds will be issued. You can increase this value much more and everything would be fine and cheaper. Try with at least 60 seconds, and increase as you see fit. 
https://api.docs.bullmq.io/interfaces/v5.WorkerOptions.html#drainDelay

**Reply by chooban** [reply]
You're right about the charging, they're only charging for "Queue Commands" but those are the 44k/day that I'm seeing. I've made the change to `drainDelay` in the last few minutes and the commands/sec has dropped off significantly. Thanks!

**Reply by sailesha** [reply]
I wanted to chime in that we're running into the same problem. I recently rearchitected our code base to have feature specific Queues and the number of commands has gone up significantly. I'll try setting drainDelay and see what happens.
![image](https://github.com/user-attachments/assets/fbd6356e-1514-487e-9468-2183e3e66342)
